### PR TITLE
Fail validation for GA packages using non-GA package spec versions

### PR DESCRIPTION
**Pull request by jsoriano**

### CODE REVIEW

**Comment by jsoriano** [comment]
test integrations

**Review by mrodm — APPROVED** [review]

**Comment by mrodm** [comment]
Running `elastic-package lint -v` would not show to the user if there any other validation errors in the package, just show the error about non-GA spec
```
 $ elastic-package lint -v 
2023/09/27 14:06:26 DEBUG Enable verbose logging
2023/09/27 14:06:26 DEBUG Distribution built without a version tag, can't determine release chronology. Please consider using official releases at https://github.com/elastic/elastic-package/releases
Lint the package
2023/09/27 14:06:26 DEBUG Check if README.md is up-to-date
2023/09/27 14:06:26 DEBUG Generate README.md file (package: /home/mariorodriguez/Coding/work/integrations/packages/elastic_package_registry)
2023/09/27 14:06:26 DEBUG Template file for README.md found: /home/mariorodriguez/Coding/work/integrations/packages/elastic_package_registry/_dev/build/docs/README.md
2023/09/27 14:06:26 DEBUG Using links definitions file: /home/mariorodriguez/Coding/work/integrations/links_table.yml
2023/09/27 14:06:26 DEBUG Render README.md file (package: /home/mariorodriguez/Coding/work/integrations/packages/elastic_package_registry, templatePath: /home/mariorodriguez/Coding/work/integrations/packages/elastic_package_registry/_dev/build/docs/README.md)
2023/09/27 14:06:26 DEBUG Read existing README.md file (package: /home/mariorodriguez/Coding/work/integrations/packages/elastic_package_registry)
Error: using non-GA Package Spec v3.0.0-next for GA package
```

Would it make sense to show also those other errors if any and fail ?

File: cmd/lint.go [hunk 1]
```diff
@@ -69,10 +71,44 @@ func validateSourceCommandAction(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return fmt.Errorf("locating package root failed: %w", err)
 	}
+
+	err = validateSpecVersion(packageRootPath)
+	if err != nil {
+		return err
+	}
+
```
Comment (mrodm) [inline cmd/lint.go:79]: Maybe, this can be moved after `validator.ValidateFromPath()` to show the linting errors that could be in the package?

**Comment by jsoriano** [comment]
Closing this and we will add this in package spec with a filtering code.

**Pull request closed without merging**